Sources/Starknet/Providers/StarknetProvider/JsonRpcParams.swift

```diff
@@ -1,10 +1,9 @@
 import Foundation
 
-typealias EmptySequence = [String]
+public typealias EmptySequence = [String]
+public struct EmptyParams: Encodable {}
```

Comment:
> It could be theoretically achieved, if we decrease params' level of abstraction.

Can we do that? I think it would considerably simplify the codebase (no unnecessary generis, no public param structs), but if that's too much effort it can be done in a separate PR.

Reply:
Yep, I've added `EncodableParams` which is a enum with possible params, there's also `EncodableParams` extension with encode method. This way we keep all param structs internal and have only one generic 👍